File: baekjoon/1253.py
n = int(input())
arr = list(map(int, input().split()))
arr.sort()
flag = 0
cnt = 0

for target in range(n):
    start = 0
    end = n-1
    while start < end:
        if arr[start] + arr[end] == arr[target]:
            if start != target and end != target:
                cnt += 1
                break
            elif start == target:
                start += 1
            elif end == target:
                end -= 1
        elif arr[start] + arr[end] > arr[target]:
            end -= 1
        elif arr[start] + arr[end] < arr[target]:
            start += 1


print(cnt)

File: baekjoon/1253.js
// https://www.acmicpc.net/problem/1253
// 좋다
// 이분탐색 투포인터

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const n = parseInt(input.shift());
const numbers = input.shift().split(" ").map(Number);

function solution() {
  let answer = 0;
  numbers.sort((a, b) => a - b);
  for (let i in numbers) {
    let target = numbers[i];
    let l = 0;
    let r = n - 1;
    while (true) {
      if (i == l) l++;
      if (i == r) r--;
      if (l >= r) break;

      let sum = numbers[l] + numbers[r];

      if (sum == target) {
        answer++;
        break;
      } else if (sum > target) {
        r--;
      } else {
        l++;
      }
    }
  }
  console.log(answer);
}
solution();
